### PR TITLE
feat(commands): add FlutterSuper and FlutterReanalyze

### PR DESCRIPTION
**Pull request by changyiyao**
FlutterSuper respects Spec: https://github.com/dart-lang/sdk/blob/main/pkg/analysis_server/tool/lsp_spec/README.md#darttextdocumentsuper-method

FlutterReanalyze respects Spec: https://github.com/dart-lang/sdk/blob/master/pkg/analysis_server/tool/lsp_spec/README.md#dartreanalyze-method.

### CODE REVIEW

File: lua/flutter-tools.lua [hunk 1]
```diff
@@ -64,6 +64,9 @@ local function setup_commands()
   cmd("FlutterSuper", function()
     require("flutter-tools.lsp").dart_lsp_super()
   end, {})
+  cmd("FlutterReanalyze", function()
```
Comment (akinsho) [inline lua/flutter-tools.lua:67]: I believe the documentation recommends not using this method. I actually historically omitted it on purpose for this reason. See the excerpt from the docs below

> Clients should not usually need to call this method - needing to do so may indicate a bug in the server.

**Review by akinsho — APPROVED** [review]
LGTM

File: lua/flutter-tools.lua (hunk 1 above)
Comment (akinsho) [reply lua/flutter-tools.lua:67]: 🤷🏿‍♂️ will leave it up to the user to use as they please and remove it if it introduces problems

**Comment by akinsho** [comment]
Thanks for the contributions @changyiyao

File: lua/flutter-tools.lua (hunk 1 above)
Comment (changyiyao) [reply lua/flutter-tools.lua:67]: > I believe the documentation recommends not using this method. I actually historically omitted it on purpose for this reason. See the excerpt from the docs below
> 
> > Clients should not usually need to call this method - needing to do so may indicate a bug in the server.

I think it is useful when I change dependency from remote to local path without quiting and reopening nvim as I'm running flutter project, which LSP could resolve symbols from local path not from remote(~/.pub-cache/)

**Pull request merged**